conformance/lang/expressions/mapping_constructor_expr.balt

```diff
@@ -0,0 +1,1882 @@
+    any var17 = q;
+    io:println(var17 is record {|readonly true myField2; any...; |}); // @output true
+
+    map<int> r = {readonly field1: MYINT};
+    io:println(r is record {|readonly 6 field1; int...; |}); // @output true
+}
+
+Test-Case: output
+Description: Test if the applicable contextually expected type is a mapping type descriptor, then that used as the 
+             inherent type with readonly specific fields.
+Labels: any, is-expr, mapping-constructor-expr, map-type, readonly-field, record-rest-descriptor, record-type, string
+
+function init() {
+    string fieldA2 = "qwerty";
+    map<string> a = {fieldA1: "String1", readonly fieldA2};
+    any var1 = a;
+    io:println(var1 is record {|readonly string fieldA2; string...; |}); // @output true
+
+    int myInt = 54;
+    record {|int field1; string field2;|} b = {readonly field1: myInt, readonly field2: getString()};
+    any var2 = b;
+    io:println(var2 is record {|readonly int field1; readonly string field2; |}); // @output true
+}
+
+function getString() returns string {
+    return "some string";
+}
+
+Test-Case: output
+Description: Test if the applicable contextually expected type is a mapping type descriptor, 
+             then that used as the inherent type in argument context
+Labels: any, array-type, is-expr, mapping-constructor-expr, map-type, module-type-defn, record-type, 
+       string, union-type
+
+type MyRecord record {|
+    int myField1;
+    boolean myField2;
+|};
+
+type MyRecord2 record {
+    int myField1;
+    boolean myField2;
+};
+
+function init() {
+    printTypeOfResult({fieldA1: "String1", fieldA2: "String2"}); // @output true
+
+    printTypeOfResult({field1: 0, field2: "mouse", field3: false}); // @output true
+
+    printTypeOfResult({field1: 0, field2: "mouse", field3: false}); // @output true
+
+    printTypeOfResult({field1: 0, field2: "mouse"}); // @output true
+
+    printTypeOfResult({field1: 23, field2: "Car", "field3": false}); // @output true
+
+    printTypeOfResult({field1: 23, field2: "Car"}); // @output true
+
+    printTypeOfResult({field1: 43, myField1: 3443, myField2: false}); // @output true
+
+    printTypeOfResult({myField1: 776777777, myField2: true}); // @output true
+
+    printTypeOfResult({field1: 0, field2: "mouse", field3: false}); // @output true
+
+    printTypeOfResult({field1: 0, field2: "mouse", "field3": false}); // @output true
+
+    printTypeOfResult({field1: 6756, field2: "cat", field3: false}); // @output true
+
+    printTypeOfResult({field1: 6756, field2: "cat", "other": "false"}); // @output true
+
+    printTypeOfResult({field1: 43, myField1: 3443, myField2: false}); // @output true
+
+    printTypeOfResult({field1: 43, myField1: 3443, myField2: false, "other": 43434}); // @output true
+
+    printTypeOfResult({myField1: 776777777, myField2: true, "other": false}); // @output true
+}
+
+function printTypeOfResult(map<any>|int[] result) {
```

Comment:
Shall we add few more tests with other types also? like `map<int>` ...

Reply:
Added in [a3ba756](https://github.com/ballerina-platform/ballerina-spec/pull/1047/commits/a3ba756463219f5c1e5b7095f194486a9028b0c9)